### ISSUE TITLE
Group tab shortcut does not work for a window with 1 parent terminal with N internal terminals 

### ISSUE THREAD

**Issue by manueljordan** [gnome-terminator/terminator#893]
Hello

**Describe the bug**
`super + t` does not work for a window with one parent terminal and at least two internal terminals. 

**To Reproduce**
Open Terminator, just keep the default parent terminal and add two internal terminals, does not matter the layout. Then execute the `super + t` shortcut. It does not work. **But** if is opened a second parent terminal _then_ the mentioned shortcut in the first parent terminal works now.

**Expected behavior**
That the `super + t` shortcut work in a window with its default parent terminal for the N internal terminals.
_**Therefore**_ is expected that the N internals terminals are grouped.

**Desktop:**
 
**One**
- OS or Linux Distribution: Fedora Workstation 39
 - Version 2.1.3
 - Display Technology: Wayland

**Two** 
 - OS or Linux Distribution: Ubuntu Desktop 22.04.4
 - Version 2.1.1
 - Display Technology: Wayland

**Additional context**
It happens for N windows with the same scenario too.

Therefore for a new user for Terminator: if he/she reaches the [Manipulating terminal groups](https://gnome-terminator.readthedocs.io/en/latest/grouping.html#manipulating-terminal-groups) section where appears a table (pls use the `Group tab` term as search key in the page) with the mentioned shortcut does not work and it causes confusion.

As an observation: Even when is true the shortcut does not work, the contextual menu works.

**Comment by rcalixte** [comment]
Hello, if I read this correctly, you're looking for the 'Group tab' functionality to work even when there are no tabs and only a single window? If that is correct, I believe that is intended. If that is incorrect, can you include screenshots that describe your issue further?

**Comment by manueljordan** [comment]
Thanks for the reply

Consider the following:

```
Window 1
 Parent Terminal A
  Internal Terminal 1A
  Internal Terminal 2A
 Parent Terminal B
  Internal Terminal 1B
  Internal Terminal 2B
```

The `super + t` shortcut works fine for either _Parent Terminal A_ or _Parent Terminal B_. Therefore: 

* is possible select _Internal Terminal 1A/2A_ for `Parent Terminal A`  

**or**

* is possible select _Internal Terminal 1B/2B_ for `Parent Terminal B` 

The point here, the `super + t` shortcut works because exists at least two Parent Terminals.

Now with the following:

```
Window 1
 Parent Terminal A
  Internal Terminal 1A
  Internal Terminal 2A
```

The `super + t` shortcut does _not_ work for _Parent Terminal A_. Therefore: 

* is **not** possible select _Internal Terminal 1A/2A_. 
 
The point here, the `super + t` shortcut does **not** work because exists only one Parent Terminal. 

And remember: _Even when is true the shortcut does not work, the contextual menu works_. Therefore with the GUI - through the little square declared in the left top of each Internal Terminal - is possible select both.

**Comment by manueljordan** [comment]
Hello 

As recap, having 1 Window with one Parent Terminal and it with two internal terminals as follows:

![Terminator-01](https://github.com/gnome-terminator/terminator/assets/2552795/1230a775-a3b4-434f-bb7a-d95ae0248102)

 Therefore as indicated `super + t` does not work. But according with the following Figure

![Terminator-02](https://github.com/gnome-terminator/terminator/assets/2552795/a61ddd68-68b3-4bb8-af5c-269a706c1ad8)

Only is possible do `Broadcast all` (or `super + g`) and it can be problematic if exists other Windows.

Now having 1 Window with at least two Parent Terminals as follows:

![Terminator-03](https://github.com/gnome-terminator/terminator/assets/2552795/cb832506-d7ab-4e55-ba05-5f8d31209892)

Now  `super + t` works and the GUI offers the _Group all in tab_ item as follows:

![Terminator-04](https://github.com/gnome-terminator/terminator/assets/2552795/2d9f164f-7a72-488b-ad9b-8f10385d2b7e)

Therefore both internal terminals are grouped

![Terminator-05](https://github.com/gnome-terminator/terminator/assets/2552795/e3d1f2c0-3b04-4a95-b174-76856dcd5b50)

> You can "Group window" or "Group all" (assuming no other windows) for them. Does that not work for you?

Really is about of the _Group all in window_. As you can see in Ubuntu does not appear that menu, but I confirmed it appears on Fedora and works as expected. Is clear the reason, it is about the version Terminator itself. _Thanks for that hint_

Let assume even if Ubuntu would has the same menu as Fedora ... why the approach is possible through the GUI and not through the shortcut?. What is the goal/reason of this difference?

Thanks for your understanding

**Comment by rcalixte** [comment]
Thanks. I think we're on the same page (including the version in Ubuntu being unfortunately out of date until 24.04 comes out next month). In picture 1, you don't have tabs open. For me, I wouldn't expect tab grouping functionality to exist unless there are tabs. (This is similar to if you used one of the key bindings to try to switch to another tab while there isn't one, e.g. Ctrl+2/Alt+2 depending on your binding. The same goes to Ctrl+PgUp/PgDn if there are no tabs. There needs to be at least one tab for tabular functionality to work. The grouping is consistent with this behavior.)

> Let assume even if Ubuntu would has the same menu as Fedora ... why the approach is possible through the GUI and not through the shortcut?. What is the goal/reason of this difference?

Do you mean that any "Group window" combination doesn't work for you when there are no tabs? There is a slight bug that has been fixed/merged (but not yet released) that addresses functionality near this but I'm not sure if it applies here. For the other issues, I would need to test in a virtual machine since I unfortunately can't replicate this (but I have other things I've done that might be the reason for that). For the "Group tab" options where there are no tabs (picture 1), I think this is expected.

If you can, it might be worth trying to use what is currently on the master branch with a caveat that it will be newer than what is in Ubuntu's current repositories. (Ubuntu's current 24.04 LTS version of terminator was last updated in March 2023.)

Thanks for the pictures! It makes it easier to point and know we're talking about the same thing. 😅

**Comment by mattrose** [comment]
Based on the conversation, I'm pretty sure this is fixed by #480  If this is not the case feel free to reopen

**Comment by mattrose** [comment]
you can upgrade to the latest terminator using either my PPA or the official Terminator PPA
